ddtrace/profiling/exporter/pprof.pyx

```diff
@@ -16,13 +16,10 @@ def _protobuf_post_312():
     # type: (...) -> bool
     """Check if protobuf version is post 3.12"""
     import google.protobuf
-    import packaging.version
+    from ddtrace.utils.version import parse_version_safe
 
-    v = packaging.version.parse(google.protobuf.__version__)
-    if isinstance(v, packaging.version.Version):
-        return v.major >= 3 and v.minor >= 12
-
-    return False
+    v = parse_version_safe(google.protobuf.__version__)
```

Comment:
@jd I modified this... I can't say that I love the addition of `parse_version_safe`, but seemed like an easy way to capture what you had (without talking to you).

the trouble with moving it to `parse_version` is that it will raise in case where you will get a LegacyVersion, which you do not have handled here. do you think it is appropriate to raise on a misformatted version here?

as well, we have a constraint mismatch, I have `packaging>=17.1`, but you require `packaging>=20`. I can adjust my requirement up if need be as well, just wanted to call it out.

Reply:
Just use `parse_version` and return `False` if it's a parsing with `Version` fails 🤷 
That'll do the same.

I don't think protobuf is going to change their version number and this code here will just be removed in a few … years. 😉 

I don't think I added a version requirement to packaging, did I? What's the feature used from version 20?

Reply:
Sorry, to confirm, you are fine with this code raising a `packaging.version.InvalidVersion` in the case where we'd get a `LegacyVersion`? Or you mean you are fine with a `try/except: return False` ?

For the feature, it is the properties: `Version.{major,minor,micro}`

Since I have `parse_version` returning a tuple, I don't require those, which is why I went with `>=17.1` as version constraint, if we think we need those properties elsewhere then I am ok with updating the requirement to >=20

Reply:
Given all this ongoing business with `Version` vs `LegacyVersion`, perhaps could be worth reconsidering https://github.com/DataDog/dd-trace-py/pull/2395#discussion_r625642977?

Reply:
yeah, worth considering. I think I'd still prefer to avoid owning the versioning parsing piece since it could change the the better changes of `packaging` keeping up to date with that. Instead I think the alternative (what we are discussing) is what is the one standard way we are going to use this library to parse versions ?

if you see my suggestion below, I think I've over complicated things a bit, and we can actually make this rather easy to go from `packaging.version.parse` -> `tuple[int, int, int]` without too much hassle (?)

https://github.com/DataDog/dd-trace-py/pull/2395#discussion_r628142671